craft_providers/lxd/launcher.py

```diff
@@ -36,6 +38,37 @@
 logger = logging.getLogger(__name__)
 
 
+class InstanceTimer(threading.Thread):
+    """Timer for update instance that still alive."""
+
+    count: int = 0
+    active: bool = True
+    instance: LXDInstance
+
+    def __init__(self, instance: LXDInstance, interval: int = 3) -> None:
+        self.instance = instance
+        self.interval = interval
+        super().__init__(daemon=True)
+
+    def run(self) -> None:
+        """Run the timer."""
+        while self.active:
+            self.count += 1
```

Comment:
Question for both you and @mr-cal about this: should we use some sort of date/time value (e.g. an ISO 8601 timestamp) instead for updating the timer? That way it becomes obvious when the timer was last updated.

Reply:
We can use time if needed, it just need to be changed every x seconds. So put a ISO 8601 should be fine.

Reply:
I like the idea of a timestamp too, I could see it being useful if we need to debug. 

The count is fine as well.